dapp/src/utils/orangekit/ledger-live/bitcoin-provider.ts

```diff
@@ -0,0 +1,322 @@
+import {
+  Account,
+  defaultLogger,
+  WalletAPIClient,
+  WindowMessageTransport,
+} from "@ledgerhq/wallet-api-client"
+import { UserRejectedRequestError } from "viem"
+import {
+  Balance,
+  BitcoinAddress,
+  BitcoinTxHash,
+  OrangeKitBitcoinWalletProvider,
+} from "@orangekit/react/src/wallet/bitcoin-wallet-provider"
+import {
+  BitcoinNetwork,
+  AcreBitcoinProvider,
+  BitcoinSignatureHelper,
+  SafeTransactionData,
+} from "@acre-btc/sdk"
+import {
+  AcreMessage,
+  AcreMessageType,
+  AcreModule,
+} from "@ledgerhq/wallet-api-acre-module"
+import BigNumber from "bignumber.js"
+
+export type TryRequestFn<T> = (fn: () => Promise<T>) => Promise<T>
+
+function tryRequest<T>(): TryRequestFn<T> {
+  return async (fn) => {
+    try {
+      return await fn()
+    } catch (error: unknown) {
+      const err = error as Error
+
+      // We want to throw the same error for all connectors when user rejects
+      // the request.
+      if (
+        err?.message === "UserRefusedOnDevice" ||
+        err?.message === "Canceled by user"
+      ) {
+        throw new UserRejectedRequestError(err)
+      }
+
+      throw error
+    }
+  }
+}
+
+export type LedgerLiveWalletApiBitcoinProviderOptions = {
```

Comment:
Let's rename it to match the provider name.
On the other hand, do we need to export it? Can't it be simply named `Options`?

```suggestion
export type AcreLedgerLiveBitcoinProviderOptions = {
```

Reply:
We use this type in `connector.ts`.

Reply:
2164f13